src/stream-ciphers.rst

```diff
@@ -255,83 +255,81 @@ Conclusion
+:term:`mode of operation`. This operation involves XORing plaintext blocks with the previous
+ciphertext block before it is encrypted by the block cipher.
+
+Of course, this process presents a problem for the first plaintext block:
+there is no previous ciphertext block to XOR it with. Instead, we select
+an IV: a random block in place of the “first” ciphertext.
+:term:`initialization vector`\s also appears in many
+algorithms. An :term:`initialization vector` should be unpredictable, ideally,
+cryptographically random. IVs do not have to be kept secret:
+they are typically just added to ciphertext messages in plaintext. It may
+sound contradictory that an IV must be unpredictable, but not
+kept a secret. It is important to remember that an attacker should not
+be capable of predicting a given IV *ahead of time*. We
 illustrate this later with an attack on predictable CBC IVs.
 
 The following diagram demonstrates encryption in :term:`CBC mode`:
 
 .. figure:: ./Illustrations/CBC/Encryption.svg
    :align: center
 
-Decryption is the inverse construction, with block ciphers in decryption
+Decryption is the inverse construction. The block ciphers are in decryption
 mode instead of encryption mode:
 
 .. figure:: ./Illustrations/CBC/Decryption.svg
    :align: center
 
-While :term:`CBC mode` itself is not inherently insecure (unlike :term:`ECB mode`), its
-particular use in TLS 1.0 was. This eventually led to the BEAST attack,
-which we'll cover in more detail in the section on SSL/TLS. The short
-version is that instead of using unpredictable :term:`initialization vector`\s,
-for example by choosing random IVs, the standard used the previous
-ciphertext block as the IV for the next message. Unfortunately, it turns
-out that attackers figured out how to exploit that property.
+While :term:`CBC mode` is not inherently insecure (unlike :term:`ECB mode`), its
+particular use was in TLS 1.0. This eventually led to the BEAST attack,
+which we detail in the SSL/TLS section. The short
+version is that instead of using unpredictable :term:`initialization vector`\s like random IVs,
+the previous ciphertext block was used as the IV for the next message. Unfortunately,
+attackers found out how to exploit this property.
 
 Attacks on CBC mode with predictable IVs
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Suppose there's a database that stores secret user information, like
-medical, payroll or even criminal records. In order to protect that
-information, the server that handles it encrypts it using a strong block
-cipher in :term:`CBC mode` with a fixed key. For now, we'll assume that that
-server is secure, and there's no way to get it to leak the key.
+Suppose there is a database storing secret user information, like medical,
+payroll or criminal records. The server protects the information by encrypting
+it with a strong block cipher in :term:`CBC mode` with a fixed key. For now, we
+assume the server is secure, and no way for a key leak.
```

Comment:
"for the key to leak" seems better, I've never heard "key leak" in this sense

Reply:
Updated just now :)